### ISSUE TITLE
Replacing all colors breaks the build

### ISSUE THREAD

**Issue by sebastiandedeyne** [tailwindcss/tailwindcss#770]
This bug occured while beta testing the new v1 config, it's not an issue in the stable release.

With the following config:

```js
module.exports = {
  theme: {
    colors: {
      blue: "#084062",
      red: "#ee585f",
      paper: "#f5f2ea"
    }
  }
};
```

The build breaks because it expects a specific shade of grey to exist:

https://github.com/tailwindcss/tailwindcss/blob/v1.0.0-beta.1/stubs/defaultConfig.stub.js#L241

```
TypeError: Cannot read property '700' of undefined
    at Array.reduce (<anonymous>)
    at new Promise (<anonymous>)
```

Currently working around the issue by extending the colors instead.

**Comment by MattStypa** [comment]
You can work around this by explicitly defining `borderColor` in your theme.

**Comment by adamwathan** [comment]
Trying to think through the best way to solve this problem at a general level, I wonder if it would make more sense for closures to receive a theme _function_ instead of a theme _object_, something that behaves just like the theme function we make available in CSS or the config function we pass to plugins, where you can use dot notation and provide a default?

Then our definition for borderColor could be something like this:

```js
    borderColor: theme => {
      return global.Object.assign({ default: theme('colors.gray.700', 'currentColor') }, theme.colors)
    },
```

**Comment by sebastiandedeyne** [comment]
On the one hand, I like an object because it opens more doors for potential auto-completion, on the other hand for cases like this a function seems to make sense. You could support both, but not sure if it's worth it to have two ways to do the same thing 🤷‍♂️

**Comment by MattStypa** [comment]
What if the `theme` passed to these closures is a `theme` object with added `get` property that provides this functionality.

```
borderColor: theme => Object.assign({ default: theme.get('colors.gray.700', 'currentColor') }, theme.colors)
```

**Comment by adamwathan** [comment]
Yeah not a bad option either 👍 I like the idea of a function because it's consistent with other areas in the framework, but it's sort of annoying we need to do anything at all just to handle this one weird default. I also like that with the object you can destructure it.

One easy option is just relying on lodash inside the default config, but that means when we let the user generate a full config they'd have lodash as a dependency in there which sucks.

Wondering if there are other options that could let us just avoid the problem altogether (hardcoding the default border color for example instead of making it dependent), or if there would be value to making it possible to do stuff like `theme('foo.bar', 'default')` for the end-user's benefit.

**Comment by adamwathan** [comment]
The other thing I like about just using a function is it makes this mistake a little more impossible — you'll always get _some_ kind of value instead of an error and it's more obvious you should provide a default in case the key you are looking for doesn't exist. Going to work on a PR for that now.

**Comment by adamwathan** [comment]
Resolved in #774, thanks for catching this.
